src/Coordination/Changelog.h

```diff
@@ -160,6 +165,11 @@ class Changelog
     /// min_log_id + 1 == max_log_id means empty log storage for NuRaft
     uint64_t min_log_id = 0;
     uint64_t max_log_id = 0;
+    /// For compaction, queue of delete not used logs
+    /// 128 is enough, even if log is not removed, it's not a problem
+    boost::lockfree::spsc_queue<std::string> del_log_q{128};
```

Comment:
@zhanglistar can you please explain -- why we need lock free queue? I cannot believe that pushing paths on log compaction into queue can lead to slow downs. Ordinary queue with mutex or concurrent bounded queue is 100% enough. `lockfree` data structure here is 100% overkill.

Reply:
Let's rename it to `log_files_to_delete_queue`

Reply:
> @zhanglistar can you please explain -- why we need lock free queue? I cannot believe that pushing paths on log compaction into queue can lead to slow downs. Ordinary queue with mutex or concurrent bounded queue is 100% enough. `lockfree` data structure here is 100% overkill.

@alesapin It's OK to use concurrent bounded queue here and I will use concurrent bounded queue because log store should be thread safe.